### ISSUE TITLE
Switch from GYB to macros/SwiftSyntaxBuilder in the standard library

### ISSUE THREAD

**Issue by CrazyFanFan** [apple/swift#68575]
Since Swift now supports macros, can we consider replacing GYB with macros to achieve the same capabilities? This would make Swift more 'Swift'.

**Comment by AnthonyLatsis** [comment]
Why have you closed this?

**Comment by CrazyFanFan** [comment]
> Why have you closed this?

My original intention was to tidy up some issues I proposed that have been left undiscussed for a long period. I assumed that the community might not consider these as good ideas.

**Comment by AnthonyLatsis** [comment]
Could you elaborate on what you meant by replacing GYB with macros? If you were referring to the metaprogrammed standard library Swift sources, then I think we can leave this open as a task, but I imagine we might eventually want to generate those using the swift-syntax SyntaxBuilder library rather than macro-expanding the declarations.

**Comment by CrazyFanFan** [comment]
Yes, what I want to replace is the GYB in the metaprogrammed Swift standard library sources.  
Additionally, using the swift-syntax SyntaxBuilder would be a good idea.

**Comment by stephentyrone** [comment]
gyb does straight textual substitution, and as such, cannot easily be replaced with macros in general.
